src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -0,0 +1,260 @@
+                  "performed per-page, this determines the maximum number of",
+                  "resources in a page."
+                });
+      } else {
+        docLines = context.getNamer().getDocLines(field);
+      }
+
+      paramDoc.firstLine(docLines.get(0));
+      paramDoc.remainingLines(docLines.subList(1, docLines.size()));
+
+      allDocs.add(paramDoc.build());
+    }
+    return allDocs;
+  }
+
+  public SimpleParamDocView getRequestObjectParamDoc(
+      MethodTransformerContext context, TypeRef typeRef) {
+    return SimpleParamDocView.newBuilder()
+        .paramName("request")
+        .typeName(context.getTypeTable().getAndSaveNicknameFor(typeRef))
+        .firstLine("The request object containing all of the parameters for the API call.")
+        .remainingLines(Arrays.<String>asList())
+        .build();
+  }
+
+  public ParamDocView getOptionalArrayParamDoc(
+      MethodTransformerContext context, Iterable<Field> fields) {
+    MapParamDocView.Builder paramDoc = MapParamDocView.newBuilder();
+
+    Name optionalArgsName = Name.from("optional", "args");
+
+    paramDoc.paramName(context.getNamer().varName(optionalArgsName));
+    paramDoc.typeName(context.getNamer().getOptionalArrayTypeName());
+
+    List<String> docLines = null;
+    if (!fields.iterator().hasNext()) {
+      String retrySettingsDocText =
+          String.format(
+              "Optional. There are no optional parameters for this method yet;\n"
+                  + "          this %s parameter reserves a spot for future ones.",
+              context.getNamer().varReference(optionalArgsName));
+      docLines = context.getNamer().getDocLines(retrySettingsDocText);
+    } else {
+      docLines = Arrays.asList("Optional.");
+    }
+    paramDoc.firstLine(docLines.get(0));
+    paramDoc.remainingLines(docLines.subList(1, docLines.size()));
+
+    paramDoc.arrayKeyDocs(getMethodParamDocs(context, fields));
+
+    return paramDoc.build();
+  }
+
+  public ParamDocView getCallSettingsParamDoc(MethodTransformerContext context) {
+    MapParamDocView.Builder paramDoc = MapParamDocView.newBuilder();
+
+    paramDoc.paramName(context.getNamer().varName(Name.from("call", "settings")));
+    paramDoc.typeName(context.getNamer().getOptionalArrayTypeName());
+    paramDoc.firstLine("Optional.");
+    paramDoc.remainingLines(new ArrayList<String>());
+
+    List<ParamDocView> arrayKeyDocs = new ArrayList<>();
+    SimpleParamDocView.Builder retrySettingsDoc = SimpleParamDocView.newBuilder();
+    retrySettingsDoc.typeName(context.getNamer().getRetrySettingsClassName());
+
+    Name retrySettingsName = Name.from("retry", "settings");
+    Name timeoutMillisName = Name.from("timeout", "millis");
+
+    retrySettingsDoc.paramName(context.getNamer().varName(retrySettingsName));
+    String retrySettingsDocText =
+        String.format(
+            "Retry settings to use for this call. If present, then\n%s is ignored.",
```

Comment:
I still think it is more preferable to rely on formatting tools since we do not know how long the line will be.  But I am okay with deferring this issue for later.